README.md

```diff
@@ -1,2 +1,788 @@
+```
+
+## Concepts
+
+Basic concepts of `SimplrFlux` are no different from [original flux](https://facebook.github.io/flux/) [concepts](https://github.com/facebook/flux/tree/master/examples/flux-concepts).
+
+We also recommend you to follow [best practices](https://facebook.github.io/flux/docs/flux-utils.html#best-practices) original flux is proposing.
+
+## Actions
+
+> Actions define the internal API of your application. They capture the ways in which anything might interact with your application.
+
+In `SimplrFlux` action is basically a class with all necessary data provided.
+
+```ts
+// action with no data provided
+export class ResetCountAction { }
+
+// action with additional data provided
+export class CountChangedAction {
+    constructor(private count: number) {}
+
+    public get Count() {
+        return this.count;
+    }
+}
+```
+
+## Dispatcher
+
+> The dispatcher receives actions and dispatches them to stores that have registered with the dispatcher.
+
+Every store will receive actions that were registered in specific store.
+
+Dispatching is usually performed in [ActionCreators](#action-creators).
+
+```ts
+import { Dispatcher } from "simplr-flux";
+import {
+    CountChangedAction
+} from "./counter-actions";
+
+export function CountChanged(count: number) {
+    Dispatcher.dispatch(new CountChangedAction(count));
+}
+```
+
+<a name="action-creators"></a>
+
+## Action creators
+
+An action creator is a set of functions to dispatch actions.
+You may choose the way to aggregate these functions,
+but we recommend you to use exported functions within single namespace when using TypeScript.
+
+```ts
+import { Dispatcher } from "simplr-flux";
+import {
+    CountUpAction,
+    CountDownAction,
+    ResetCountAction,
+    CountChangedAction
+} from "./counter-actions";
+
+export namespace CounterActionsCreators {
+    export function CountUp() {
+        Dispatcher.dispatch(new CountUpAction);
+    }
+
+    export function CountDown() {
+        Dispatcher.dispatch(new CountDownAction);
+    }
+
+    export function ResetCount() {
+        Dispatcher.dispatch(new ResetCountAction);
+    }
+
+    export function CountChanged(count: number) {
+        Dispatcher.dispatch(new CountChangedAction(count));
+    }
+}
+```
+
+## Stores
+
+> A store is what holds the data of an application. Stores will register with the application's dispatcher so that they can receive actions.
+
+Main difference between store types in `SimplrFlux` is state management.
+
+In accordance with your data structure you may choose type of store that best fits your data structure.
+
+### ReduceStore
+
+ReduceStore `state` can be any `object` and have no structure constraints.
+
+Only way to change `state` of `ReduceStore` is actions `handlers`.
+Actions are registered in store's `constructor` using protected method [`registerAction`](#register-action) which takes `action` class and `handler` function as arguments (check [`API`](#reduce-store-api) section).
```

Comment:
-> Actions are registered in store's `constructor` using protected method [`registerAction`](#register-action) which takes an `action` class and a `handler` function as arguments (check [`API`](#reduce-store-api) section).